### PR TITLE
PaX MPROTECT integration

### PR DESCRIPTION
**Pull request by carenas**
tested in NetBSD 9.0 amd64, NetBSD 8.2 i386 and NetBSD 7.2 i386

CC: @krytarowski

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ EXAMPLE_TARGET = $(BINDIR)/func_call $(BINDIR)/first_program $(BINDIR)/branch $(
 SLJIT_HEADERS = $(SRCDIR)/sljitLir.h $(SRCDIR)/sljitConfig.h $(SRCDIR)/sljitConfigInternal.h
 
 SLJIT_LIR_FILES = $(SRCDIR)/sljitLir.c $(SRCDIR)/sljitUtils.c \
-	$(SRCDIR)/sljitExecAllocator.c $(SRCDIR)/sljitProtExecAllocator.c \
+	$(SRCDIR)/sljitExecAllocator.c $(SRCDIR)/sljitProtExecAllocator.c $(SRCDIR)/sljitWXExecAllocator.c \
```
Comment (zherczeg) [inline Makefile:35]: This is missing from my patch. I am sorry.

Comment (carenas) [reply Makefile:35]: feel free to squash this chunk and I'll rebase over, it is here mainly to easy testing

**Comment by krytarowski** [comment]
The NetBSD specific integration is looking good.

**Comment by carenas** [comment]
> The NetBSD specific integration is looking good.

the W^X allocator seems to work fine in Linux armv8l (Debian sid armel) using ARMv5 which I got through crostini/lxc in an arm64 chromebook

did you get any luck with sparc?; I suspect TileGX might not, but that is unlikely to be a regression

**Comment by krytarowski** [comment]
Sparc should just work.. but if there are concerns I propose to add dedicated sljit-independent ATF NetBSD tests for all architectures and cover sparc too.

Once the patch will stabilize, I will spin a sparc virtual machine.

**Comment by carenas** [comment]
you had concerns with alignment issues and sparc are the best canary architecture for those kind of problems since they don't handle misaligned access in hardware at all, and so will trap and sometimes crash.

I don't think the patch will have any significant changes might warrant waiting otherwise

**Comment by carenas** [comment]
probably needs #63 and other changes like  72ca72c8e22f13d07b1b4125d70e3f07587f51c7 to be "complete", plus github's  automerge looks like a bad idea for repository history so it might be better rebasing this or even resubmitting.

feedback and test reports encouraged

**Pull request closed without merging**